### PR TITLE
Add MustGet functionality to Memcache

### PR DESCRIPTION
**Pull request by AndyHaskell**
Problem: If `Memcache.Get` errors on its initial memcache retrieval attempt, but the error it gets was not `ErrCacheMiss` (for example, if memcache is temporarily unavailable), an error is returned without attempting the `RetrieveFunc` passed in

Fix: For the memcache Cacher, add a `MustGet` method for attempting the `RetrieveFunc` if the initial retrieval errors regardless of whether or not the error that occurred in the initial memcache retrieval was `ErrCacheMiss` for greater flexibility. Additionally, this behavior in `Get` is now documented

### CODE REVIEW

**Comment by kujenga** [comment]
@AndyHaskell Just a thought for an alternative implementation, what do you think about adding this as an option in a struct field on `Memcache`? That way we don't have to break out of the `Cacher` interface for a problem specific to this cacher.

**Comment by AndyHaskell** [comment]
Good idea!

**Comment by kujenga** [comment]
This looks good! My one additional concern is that if a memcache retrieval fails, then a memcache lock will probably fail as well, so this just gets us further down in the method. Perhaps more error handling is necessary?

File: caching/memcache.go [hunk 1]
```diff
@@ -62,8 +69,25 @@ func (m *Memcache) Get(k interface{}, target interface{}, fn RetrieveFunc) error
 	ctx, _ := appengine.Namespace(m, cacheNamespace)
 
 	if _, err := memcache.Gob.Get(ctx, key, target); err != memcache.ErrCacheMiss {
-		// either nil (success), or unknown failure
-		return err
+		if m.MustGet && err != nil {
+			// If MustGet is true and the error we got was a Memcache is temporarily unavailable
+			// error, then it's unlikely that we will be able to acquire a memcache lock, so in
+			// that case, we just attempt to call the fallback RetrieveFunc to retrieve our data
+			if strings.Contains(err.Error(), "Memcache is temporarily unavailable.") {
```
Comment (kujenga) [inline caching/memcache.go]: Is there anything more precise we can do here? We might want to use `memcache.ErrServerError` here as well: https://godoc.org/google.golang.org/appengine/memcache#pkg-variables

Comment (AndyHaskell) [inline caching/memcache.go]: I'd say `ErrServerError` is fitting to check

**Comment by kujenga** [comment]
Great, this looks good to me 👍

**Comment by kujenga** [comment]
Hmm, just had another thought, do we want to log out a warning or anything? Having this happen silently is potentially scary.

**Pull request merged**